src/main/java/org/candlepin/insights/pinhead/PinheadService.java

```diff
@@ -52,13 +59,32 @@ public PinheadService(PinheadApiProperties apiProperties, PinheadApi api,
         this.retryTemplate = retryTemplate;
     }
 
-    public OrgInventory getPageOfConsumers(String orgId, String offset) throws ApiException {
+    /**
+     * Return a page of consumers for the given org using the offset provided and with consumers that have
+     * checked in since the lastCheckinTime.  This function accepts the last check in time as a String
+     * parameter in order to validate the format prior to sending the request over the wire.  The format is
+     * similar to ISO 8601 format but the code on the other end is matching to the regular expression we're
+     * using for validation rather than doing full ISO 8601 parsing.  Therefore, things like timezone
+     * offsets are not acceptable for the request even though they are valid ISO 8601 formats.
+     * @param orgId organization id
+     * @param offset offset in the listing of results
+     * @param lastCheckinTime in the form YYYY-MM-DDTHH-MM-SSZ (e.g. 2012-12-18T23:56:23Z).
+     * @return an OrgInventory object with the relevant results
+     */
+    public OrgInventory getPageOfConsumers(String orgId, String offset,
+        @Pattern(regexp = "^([12]\\d{3}-(0[1-9]|1[0-2])-(0[1-9]|[12]\\d|3[01])T(0[0-9]|1[0-9]|2[1-4]):" +
+        "(0[0-9]|[1-5][0-9]):(0[0-9]|[1-5][0-9]))Z$") String lastCheckinTime) throws ApiException {
         return retryTemplate.execute(context -> {
             log.debug("Fetching page of consumers for org {}.", orgId);
-            OrgInventory consumersForOrg = api.getConsumersForOrg(orgId, batchSize, offset);
+            OrgInventory consumersForOrg = api.getConsumersForOrg(orgId, batchSize, offset, lastCheckinTime);
             log.debug("Consumer fetch complete. Found {} for batch of {}.", consumersForOrg.getFeeds().size(),
                 batchSize);
             return consumersForOrg;
         });
     }
+
+    public OrgInventory getPageOfConsumers(String orgId, String offset) throws ApiException {
+        String lastCheckIn = DateTimeFormatter.ISO_INSTANT.format(Instant.now());
```

Comment:
Can you please rework this to use
```suggestion
        String lastCheckIn = DateTimeFormatter.ISO_INSTANT.format(Instant.now().minus(hostCheckinThreshold));
```
where `hostCheckinThreshold` is a `Duration` in `ApplicationProperties`?

Reply:
Can do.

Reply:
@kahowell just to verify you're wanting the value here from InventoryServiceProperties.getHostLastSyncThreshold?

Reply:
Indeed.